### ISSUE TITLE
Can open_flamingo handle image classification tasks？

### ISSUE THREAD

**Issue by RayL0221** [mlfoundations/open_flamingo#237]
If I have a batch of unlabeled image data, can I handle image classification tasks using similarity scores in the Open_Flamingo framework, similar to CLIP?

Such as the code follows:

import torch
from PIL import Image
import open_clip

model, _, preprocess = open_clip.create_model_and_transforms('ViT-B-32', pretrained='laion2b_s34b_b79k')
tokenizer = open_clip.get_tokenizer('ViT-B-32')

image = preprocess(Image.open("CLIP.png")).unsqueeze(0)
text = tokenizer(["a diagram", "a dog", "a cat"])

with torch.no_grad(), torch.cuda.amp.autocast():
    image_features = model.encode_image(image)
    text_features = model.encode_text(text)
    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)

    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)

print("Label probs:", text_probs)  # prints: [[1., 0., 0.]]

Thanks！

**Comment by anas-awadalla** [comment]
Yes! We have this implemented for ImageNet and HatefulMemes [here](https://github.com/mlfoundations/open_flamingo/blob/f93ad8494ca2c6da9c388ad649ac04e73635c5a5/open_flamingo/eval/evaluate.py#L1118).
